premake_customizations.lua

```diff
@@ -0,0 +1,20 @@
+-- Implement the workspace files command for solution-scope files
+require('vstudio')
+premake.api.register {
+	name = "workspace_files",
+	scope = "workspace",
+	kind = "list:string",
+}
+
+premake.override(premake.vstudio.sln2005, "projects", function(base, wks)
+	if wks.workspace_files and #wks.workspace_files > 0 then
+		premake.push('Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{' .. os.uuid("Solution Items:"..wks.name) .. '}"')
```

Comment:
Where does this UUID come from, exactly?

Reply:
It's a GUID, and it's the one for the "solution scope". I do feel like defining it as a constant and then using that constant instead would make more sense. I found a list of the GUID in VS [here](https://www.codeproject.com/Reference/720512/List-of-Visual-Studio-Project-Type-GUIDs).